### ISSUE TITLE
PostGRESQL dialect - position function parsing error when table reference is used

### ISSUE THREAD

**Issue by AndriesDraux** [sqlfluff/sqlfluff#1969]
## SQL Query

````
select
  u.user_id,
  u.user_email,
  p.product_id
from {{ ref('user') }} as u
inner join {{ ref('product') }} as p on u.user_id = p.user_id
  and position('@domain' in u.user_email) = 0

````

## Expected Behaviour

All Finished 📜 🎉! 

## Observed Behaviour

L:  33 | P:  91 |  PRS | Line 27, Position 91: Found unparsable section:
                       | '.user_email'

## Steps to Reproduce

no parsing error: position('@randomemail' in user_email) = 0
parsing error: position('@randomemail' in u.user_email) = 0

In the parsing error part; `u` is a table alias defined in the from statement: `from users as u`

## Dialect

PostGRESQL

## Version
sqlfluff, version 0.8.1
Python 3.9.7

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = dbt
rules = None
exclude_rules = L031
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}
star = {% macro star(from,relation_alias,except,prefix) %}{{relation_alias}}.*{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 2
max_line_length = 150
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = unqualified
unquoted_identifiers_policy = all

[sqlfluff:templater:dbt]
profiles_dir = ./.dbt

# Some rules have their own specific config.
# Check all rules (including explanations) on https://docs.sqlfluff.com/en/stable/rules.html
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = lower

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:L016]
ignore_comment_lines = True

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = lower

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = lower

[sqlfluff:rules:L042] # both = not in from & join
forbid_subquery_in = both

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
```

**Comment by tunetheweb** [comment]
Can you please provide the SQL and SQLFluff config

**Comment by AndriesDraux** [comment]
Ok, added, apologies :)

**Comment by tunetheweb** [comment]
Fix submitted in #1971 and should be in next release.
